### PR TITLE
[JENKINS-50328] [JENKINS-51066] Fix bug where pipeline editor ignores custom Jenkinsfile script path.

### PR DESCRIPTION
**Pull request by sutrosoftware**
# Description
This change fixes a bug where the pipeline editor ignores and therefore fails to load or save custom Jenkinsfile script paths. This is a fix to existing functionality so no new tests have been added.

See [JENKINS-50328](https://issues.jenkins-ci.org/browse/JENKINS-50328).
See [JENKINS-51066](https://issues.jenkins-ci.org/browse/JENKINS-51066).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by sutrosoftware** [comment]
The test failure appears to be due to a jenkins restart mid-test and unrelated to the change.

**Comment by kshultzCB** [comment]
Thanks for the PR! 

As for the question of adding tests, it might be good to add something to the acceptance tests to run through this. I don't yet know what would be involved in that.

> The test failure appears to be due to a jenkins restart mid-test and unrelated to the change.

I just [relaunched](https://ci.blueocean.io/blue/organizations/jenkins/blueocean/detail/PR-2049/2/pipeline) the build.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineImpl.java [hunk 1]
```diff
@@ -255,7 +271,14 @@ public Link getLink() {
         @Override
         public MultiBranchPipelineImpl getPipeline(Item item, Reachable parent, BlueOrganization organization) {
             if (item instanceof MultiBranchProject) {
-                return new MultiBranchPipelineImpl(organization, (MultiBranchProject) item);
+                MultiBranchPipelineImpl mbpi = new MultiBranchPipelineImpl(organization, (MultiBranchProject) item);
+                if (item instanceof WorkflowMultiBranchProject) {
+                    WorkflowMultiBranchProject wfmbp = (WorkflowMultiBranchProject)item;
+                    WorkflowBranchProjectFactory wbpf = (WorkflowBranchProjectFactory)wfmbp.getProjectFactory();
```
Comment (arogge) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineImpl.java:277]: This seems to break "Multibranch with defaults" plugin as its PipelineBranchDefaultsProjectFactory cannot be cast to WorkflowBranchProjectFactory which leads to a ClassCastException.

Comment (Asgoret) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineImpl.java:277]: @kshultzCB Can you look at this plz? :)

Comment (glfp) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineImpl.java:277]: I confirm ... it break Multibranch With Defaults

**Comment by sutrosoftware** [comment]
Will take a look. Thanks for the bug report.

**Pull request merged**